umbral-pre/src/dem.rs

```diff
@@ -59,20 +64,20 @@ impl UmbralDEM {
         ciphertext: impl AsRef<[u8]>,
         authenticated_data: &[u8],
     ) -> Option<Box<[u8]>> {
-        let nonce_size = <<ChaCha20Poly1305 as AeadInPlace>::NonceSize as Unsigned>::to_usize();
+        let nonce_size = <NonceSize as Unsigned>::to_usize();
         let buf_size = ciphertext.as_ref().len();
 
         if buf_size < nonce_size {
             return None;
         }
 
-        let nonce = Nonce::from_slice(&ciphertext.as_ref()[..nonce_size]);
+        let nonce = XNonce::from_slice(&ciphertext.as_ref()[..nonce_size]);
         let payload = Payload {
             msg: &ciphertext.as_ref()[nonce_size..],
             aad: authenticated_data,
         };
         self.cipher
-            .decrypt(&nonce, payload)
+            .decrypt(nonce, payload)
```

Comment:
Just curious, was this change due to moving from chacha20 to xchacha20?

Reply:
Good question, actually no, it was an independent style fix. `nonce` is created with `GenericArray::from_slice()` which returns a reference to `GenericArray`. `decrypt()` takes a reference to `GenericArray` as the first parameter. So just passing `nonce` is enough.

But there's a bit of Rust's syntactic sugar that "compresses" a reference to a reference to just a reference. That's why passing `&nonce` (of type `&&GenericArray`, that is a reference to a reference) worked just as well.